device/transport/mqtt/src/mqtt.ts

```diff
@@ -36,21 +40,36 @@ export class Mqtt extends EventEmitter implements Client.Transport, StableConnec
   protected _config: ClientConfig;
   private _mqtt: MqttBase;
   private _twinReceiver: MqttTwinReceiver;
+  private _topicTelemetryPublish: string;
+  private _topicMessageSubscribe: string;
+  private _receiver: MqttReceiver;
 
   /**
    * @private
    */
-  constructor(config: ClientConfig, provider?: MqttBase) {
+  constructor(config: ClientConfig, provider?: any) {
     super();
     this._config = config;
+    this._topicTelemetryPublish = 'devices/' + this._config.deviceId + '/messages/events/';
+    this._topicMessageSubscribe = 'devices/' + this._config.deviceId + '/messages/devicebound/#';
+    debug('topic publish: ' + this._topicTelemetryPublish);
+    debug('topic subscribe: ' + this._topicMessageSubscribe);
+    const sdkVersionString = encodeURIComponent('azure-iot-device/' + packageJson.version);
```

Comment:
>packageJson.version [](start = 70, length = 19)

This is the version of the mqtt transport, not the version of the sdk.  Is this what we want?

Reply:
Right now they are identical. TBH we probably want a combination of transport and client versions. I think Olivier and Yi have something in the works to improve telemetry so since this is a small change that is kindof out of scope for now, I'd say let's wait and fix this later. the most important part (azure-iot-device) is correct.

---
In reply to: [141152380](https://github.com/Azure/azure-iot-sdk-node/pull/90#discussion_r141152380) [](ancestors = 141152380)